webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,279 @@
+
+export type Props = {
+    href: string;
+    show: boolean;
+    connected: boolean;
+    isLoaded?: boolean;
+    ticketDetails?: TicketDetails;
+    defaultUserInstanceID?: string;
+    connectedInstances: Instance[];
+    getIssueByKey: (issueKey: string, instanceID: string) => (dispatch: Dispatch, getState: any) => Promise<{
+        data?: TicketData;
+        error?: any;
+    }>;
+    setTicket?: (ticketDetails: {}) => void;
+}
+
+export type State = {
+    href: string;
+    isLoaded: boolean;
+    ticketId: string;
+    ticketDetails?: TicketDetails;
+};
+
+export default class TicketPopover extends React.PureComponent<Props, State> {
+    truncateString(str: string, num: number) {
+        if (num > str.length) {
+            return str;
+        }
+        return `${str.substring(0, num)}...`;
+    }
+
+    constructor(Props: Props) {
+        super(Props);
+        let ticketID = '';
+        if (this.props.href.includes('selectedIssue')) {
+            ticketID = this.props.href.split('selectedIssue=')[1].split('&')[0];
+        }
+
+        if (ticketID === '' && this.props.href.includes('atlassian.net/browse')) {
+            ticketID = this.props.href.split('|')[0].split('?')[0].split('/browse/')[1];
+        }
+
+        this.state = {
+            href: this.props.href,
+            isLoaded: false,
+            ticketId: ticketID,
+        };
+    }
+
+    async init() {
+        let instanceID = '';
+        if (this.props.connectedInstances.length === 1) {
+            instanceID = this.props.connectedInstances[0].instance_id;
+        } else if (this.props.defaultUserInstanceID) {
+            instanceID = this.props.defaultUserInstanceID;
+        }
+
+        let ticketID = '';
+        if (this.props.href.includes('selectedIssue')) {
+            ticketID = this.props.href.split('selectedIssue=')[1].split('&')[0];
+            this.props.getIssueByKey(ticketID, instanceID);
+        }
+
+        if (ticketID === '' && this.props.href.includes('atlassian.net/browse')) {
+            ticketID = this.props.href.split('|')[0].split('?')[0].split('/browse/')[1];
+            if (ticketID) {
+                this.props.getIssueByKey(ticketID, instanceID);
+            }
+        }
+    }
+
+    componentDidMount() {
+        if (this.props.connected && !this.state.isLoaded && ((this.props.ticketDetails && this.props.ticketDetails.ticketId !== this.state.ticketId) || !this.props.isLoaded)) {
+            this.init();
+        } else if (this.props.connected && !this.state.isLoaded && this.props.ticketDetails && this.props.ticketDetails.ticketId === this.state.ticketId) {
+            this.setTicket(this.props);
+        }
+    }
+
+    componentDidUpdate(): void {
+        if (this.props.isLoaded && !this.state.isLoaded && this.props.ticketDetails && this.props.ticketDetails.ticketId === this.state.ticketId) {
+            this.setTicket(this.props);
+        }
+    }
+
+    setTicket(data: Props): void{
```

Comment:
```suggestion
    setTicket(data: Props) {
```
What's the need for specifying the return type here?